lib/sntp/sntp.cc

```diff
@@ -334,6 +334,10 @@ namespace
 	{
 		static FlagLockPriorityInherited lock;
 		NetworkContext                   udpContext{timeout, nullptr};
+		SntpStatus_t                     status;
+		// Use the do {...} while {false} to deduplicate
+		// error-handling.
```

Comment:
IMHO, don't be afraid to use `goto`s for this kind of control flow.  I know that's not a universal opinion.

Reply:
Agree! Happy to move to a goto if you would prefer that. @davidchisnall any opinion?

Reply:
ACK, should we merge this as it is now then?